### PR TITLE
feat: crouching will make you prefer lockpicking over prying, fix being unable to lockpick metal doors with unusable prying tools in inventory

### PR DESCRIPTION
**Pull request by chaosvolt**
### Required

- [X] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [X] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [X] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [X] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

This improves the logic behind prying vs lockpicking doors to fix cases where having an inferior prying tool  completely locks you out of trying to pick a lock, while also adding a method Kheir suggested for a reasonably intuitive way to pick lockpicking as a first resort on command.

Fixes https://github.com/cataclysmbnteam/Cataclysm-BN/issues/5028

## Describe the solution

C++ changes:
1. In iexamine.cpp, updated `iexamine::locked_object` with some improvements. First, before the usual attempts to pry the door open, if opens with a separate check for lockpicking if the player is currently crouched, so sneaking about will automatically make you attempt stealth first if you have both tools available. Then, the prying check now also does a quick check for how hard the target of your prying attempt is to pry, just to check ahead of time whether you lack a tool that can even attempt to pry the target, and skipping the attempt if so. This is done because the normal check for prying requirement is buried in the iuse for the crowbar, so by that point the player is already committed and isn't given the option to fall back on lockpicking if they (for example) have a crowbar and lockpicks but are faced with a metal door that would need a halligan bar to pry open. Lastly, the failure messages have been reworded so that, if you've reached that point with a prying tool in your possession (meaning it was not strong enough to even attempt to pry that terrain), the message acknowledges you need better tools rather than assuming you lack a prying tool entirely.

JSON changes:
1. Updated the descriptions of lockpicking items to include a mention that crouching will have you pick them first.
2. Added a mention of it to main menu tips as well.

## Describe alternatives you've considered

Writing a function for a common pry quality threshold check for both iexamine and iuse to share.

## Testing

1. Checked affected JSON files for syntax and lint errors.
2. Compiled and load-tested.
3. Spawned in a crowbar and a locksmithing kit.
4. Set myself up with a locked wooden door and a locked metal door.
5. As expected, when standing I'll default to using the crowbar on the wooden door, and the lockpick on the metal door.
6. Crouching has me switch to using the lockpick for both doors, as intended.
7. Swapping in a halligan bar has me correctly default to using it on either door when standing, the lockpick when crouching.
8. Trying to examine the metal door with only the crowbar correctly has it say I could pry it open with stronger tools.
9. Installed fingerpick CBM, as expected I swap between it and a prying tool accordingly when crouched.
10. Checked affected C++ file for astyle.

### CODE REVIEW

**Comment by RoyalFox2140** [comment]
Did you ensure that worn hairpins won't be chosen for lockpicking which will immediately damage them?

**Comment by scarf005** [comment]
> feat: crouching will make you prefer lockpicking over prying, fix being unable to lockpick metal doors with unusable prying tools in inventory

~~_this screams two separate PRs_~~

**Comment by RoyalFox2140** [comment]
Actually, cancel that request. I may just remove lockpicking quality from hairpins as improvised lockpicks have 3 pick quality, are made from 1 scrap metal in 10 minutes with a rock.

Hairpins have either 1 or 3 lockpicking. Utterly useless IMO.

**Comment by scarf005** [comment]
>  I may just remove lockpicking quality from hairpins as improvised lockpicks have 3 pick quality, are made from 1 scrap metal in 10 minutes with a rock.
>
> Hairpins have either 1 or 3 lockpicking. Utterly useless IMO.

maybe how about we bump hairpin instead? great RP opportunity IMO

**Comment by RoyalFox2140** [comment]
Fancy and gold hairpins are too valuable to be destroyed on lockpicking attempts that happen automatically, I'd either prefer not having them be lockpicks or the game not choosing them if they're being worn by the player.

**Comment by chaosvolt** [comment]
> ~_this screams two separate PRs_~

I figured it'd be okay to have in one PR because it's a very basic if statement in the same PR.

Hairpin changes in any case seemed out of scope for this PR since that'd be changes to different C++ or JSON entirely.

> Fancy and gold hairpins are too valuable to be destroyed on lockpicking attempts that happen automatically, I'd either prefer not having them be lockpicks or the game not choosing them if they're being worn by the player.

They aren't valuable in the slightest except as improvised lockpicks though, outside of a tiny morale bonus for stylish characters or for a bit of gold if you're playing with Arcana.

If you want to keep hairpins from being destroyed, maybe a viable option would be to remove lockpicking quality from them but then make them all count as components in makeshift lockpicks, or a separate "hairpin lockpick" item in the event we want them to have a different quality or for the sake of item weight?

**Comment by RoyalFox2140** [comment]
> > ~_this screams two separate PRs_~
> 
> I figured it'd be okay to have in one PR because it's a very basic if statement in the same PR.
> 
> Hairpin changes in any case seemed out of scope for this PR since that'd be changes to different C++ or JSON entirely.
> 
> > Fancy and gold hairpins are too valuable to be destroyed on lockpicking attempts that happen automatically, I'd either prefer not having them be lockpicks or the game not choosing them if they're being worn by the player.
> 
> They aren't valuable in the slightest except as improvised lockpicks though, outside of a tiny morale bonus for stylish characters or for a bit of gold if you're playing with Arcana.
> 
> If you want to keep hairpins from being destroyed, maybe a viable option would be to remove lockpicking quality from them but then make them all count as components in makeshift lockpicks, or a separate "hairpin lockpick" item in the event we want them to have a different quality or for the sake of item weight?

Late to this since I was sleeping and didn't come back to this immediately. I think the best course of action is to allow bone skewers to have 1 lockpicking, and for all hairpins to convert to a new makeshift lockpick without a tool and in seconds. And to of course remove all lockpicking from hairpins. That said 1 scrap metal for 10 minutes with any hammer is also trivial.

**Review by scarf005 — DISMISSED** [review]

File: src/iexamine.cpp [hunk 1]
```diff
@@ -1462,23 +1462,46 @@ void iexamine::locked_object( player &p, const tripoint &examp )
...
-            add_msg( m_info, _( "The %s is locked.  If only you had something to pry it or pick its lock…" ),
-                     here.has_furn( examp ) ? here.furnname( examp ) : here.tername( examp ) );
+            if( prying_tool ) {
+                add_msg( m_info,
+                         _( "The %s is locked.  If only you had something to pick its lock, or a stronger prying tool…" ),
+                         here.has_furn( examp ) ? here.furnname( examp ) : here.tername( examp ) );
+            } else {
+                add_msg( m_info, _( "The %s is locked.  If only you had something to pry it or pick its lock…" ),
+                         here.has_furn( examp ) ? here.furnname( examp ) : here.tername( examp ) );
+            }
```
Comment (scarf005) [inline src/iexamine.cpp:1493]:
```suggestion
            const auto target = here.has_furn( examp ) ? here.furnname( examp ) : here.tername( examp );
            if( prying_tool ) {
                add_msg( m_info,
                         _( "The %s is locked.  If only you had something to pick its lock, or a stronger prying tool…" ), target );
            } else {
                add_msg( m_info, _( "The %s is locked.  If only you had something to pry it or pick its lock…" ), target );
            }
```

File: src/iexamine.cpp [hunk 2]
```diff
@@ -1462,23 +1462,46 @@ void iexamine::locked_object( player &p, const tripoint &examp )
...
 
-    add_msg( m_info, _( "The %s is locked.  If only you had something to pry it…" ),
-             here.has_furn( examp ) ? here.furnname( examp ) : here.tername( examp ) );
+    if( prying_tool ) {
+        add_msg( m_info, _( "The %s is locked, and your tools aren't strong enough to pry it open…" ),
+                 here.has_furn( examp ) ? here.furnname( examp ) : here.tername( examp ) );
+    } else {
+        add_msg( m_info, _( "The %s is locked.  If only you had something to pry it…" ),
+                 here.has_furn( examp ) ? here.furnname( examp ) : here.tername( examp ) );
+    }
```
Comment (scarf005) [inline src/iexamine.cpp:1504]: same applies to here too

**Review by scarf005 — APPROVED** [review]

**Pull request merged**